sendgrid/helpers/mail/validators.py

```diff
@@ -0,0 +1,68 @@
+################################################################
+# Various types of Validators
+################################################################
+
+class ValidateAPIKey(object):
+    """Validates content to ensure SendGrid API key is not present"""
+
+    regexes = None
+
+    def __init__(self, regex_strings=list(), use_default=True):
+        """Constructor
+        Args:
+            regex_strings (list<str>): list of regex strings
+            use_default (bool): Whether or not to include default regex
+        """
+
+        import re
+        self.regexes = set()
+
+        #Compile the regex strings into patterns, add them to our set
+        for regex_string in regex_strings:
+            self.regexes.add(re.compile(regex_string))
+
+        if use_default:
+            default_regex_string = 'SG\.[0-9a-zA-Z]+\.[0-9a-zA-Z]+'
+            self.regexes.add(re.compile(default_regex_string))
+
+
+    def validate_message_dict(self, request_body):
+        """With the JSON dict that will be sent to SendGrid's API, 
+            check the content for SendGrid API keys - throw exception if found
+        Args:
+            request_body (:obj:`dict`): message parameter that is
+                                            an argument to: mail.send.post()
+        Raises:
+            APIKeyIncludedException: If any content in request_body matches regex
+        """
+
+        #Handle string in edge-case
+        if isinstance(request_body, str):
+            self.validate_message_text(request_body)
+
+        #Default param
+        elif isinstance(request_body, dict):
+            if "content" in request_body:
+                contents = request_body["content"]
+
+                for content in contents:
+                    if "value" in content and "type" in content:
+                        if content["type"] == "text/html" or isinstance(content["value"], str):
```

Comment:
This line exceeds the 79 char limit of PEP8. Two things could help this:
* [EAFP](https://docs.python.org/3/glossary.html#term-eafp) instead of LBYL
    You could use a `try/except` instead of checking `if "value" in content` but a nicer way is just to use `content.get('value')`. This gets rid of one level of indentation immediately.
* You could then use brackets for line continuation of the second conditional check
    > [The preferred way of wrapping long lines is by using Python’s implied line continuation inside parentheses, brackets and braces. Long lines can be broken over multiple lines by wrapping expressions in parentheses. These should be used in preference to using a backslash for line continuation.](http://pep8.org/#maximum-line-length)


Maybe end up with something along these lines?

```python
for content in contents:
    if (content.get("type") == "text/html" or
            isinstance(content.get("value"), str)):
        message_text = content["value"]
        self.validate_message_text(message_text)
```

Reply:
Awesome - just implemented it in a similar fashion. That was a good idea - I'm always looking to learn how to make my code better :)